src/AgentVisibilityContext.tsx

```diff
@@ -69,13 +73,18 @@ export const VisibilityProvider = ({
     setHiddenAgents(newHidden);
   };
 
+  const handleAgentCheckboxChange = (agentName: string) => {
+    setHiddenAgents(getNewHiddenAgents(agentName, hiddenAgents));
```

Comment:
The usage here and in the tests makes me think that `getNewHiddenAgents` could be renamed to `toggleAgentVisibility`...

Reply:
If you want a glimpse into the future, these functions and handlers are probably going to take care of both hiding and highlighting and be called `handleSelectionChange` and `getNewSelectionMap`. In reference to selection interface.
[This draft PR ](https://github.com/simularium/nbsv/pull/30/files#diff-386adbd553aaff92478104f6a5c9a4b72d6570af010c0d7897c2f190ca20c2fc)is third in line and shows the approach I'm heading towards, I need these stops along the way in order to keep the PR scope smaller.

What do you think of that naming?

Reply:
That seems to be fine! I don't know if I'd don't get an intuitive understanding of what `getNewSelectionMap` means if I was looking at this codebase for the first time. I still think the `toggle` keyword is useful, but this can all be discussed in the future PRs.

Reply:
Ok, thank you for this context, I want it to be intuitive, and @meganrm steered me back towards emphasizing either the checkbox or the fact that it is a user selection. I'll ponder... maybe `getNewVisibilitySelectionMap` and something like `toggleSelectionCheckbox`